### DISCUSSION TITLE
We definetly need an offline DB

### DISCUSSION THREAD

**Discussion by NitroNerd** [d8ahazard/sd_dreambooth_extension#999] [Ideas]
A saw lots of complaints, where eolpe say, that DB must be offline. And i totally agree with this. All requirement should be updating by "update" button or something like this. If someone knows how to  force DB to work offline now, give your advice.

**Comment by Arilziem** [comment]
It is offline. It needs internet when updating (after pressing the update button in the webui extension management) and on first run (to download some models and requirements), but other than that, training does not require internet access.

**Reply by NitroNerd** [reply]
[Arilziem](https://github.com/Arilziem), just try to block connections through firewall, or turn off internet, you will understand then, that DB actually requires connection to work properly,

(requires connection at launch of automatic)

**Reply by Arilziem** [reply]
Well yes, you need that connection once so it can install all dependencies. You need to launch it with the dreambooth .bat/.sh, let it install stuff and that should be it. After that, the only thing it will do is to check for new packages, which in the past did stall out things for a bit without internet. This can be disabled though:

> And last, if you wish to completely skip the "native" install routine of Dreambooth, you can set the following environment flag: DREAMBOOTH_SKIP_INSTALL=True
> 
> This is ideal for "offline mode", where you don't want the script to constantly check things from pypi.

(from readme)

This should not happen with more recent versions of the webui though (I guess the timeout was reduced), so you don't really need that launch option anymore (but it also can't hurt unless you want to update).

And I know it works because I have tried it with the most recent version of the extension.

**Reply by NitroNerd** [reply]
[Arilziem](https://github.com/Arilziem), man you did well. The only thing left is to publish your "automatic+dreambooth portable", which working in offline mode. You'll become a hero of internet ai segment.

**Comment by Arilziem** [comment]
Joking aside, if you need help setting up offline, just come back here with the issue you are having, and I or someone else will probably be able to help.

The basic steps to get things running are (with internet access):
1. Install the webui and run it
2. Install the dreambooth extension
3. Close the webui completely
4. Copy webui-dreambooth-user.bat/sh into main folder and run it
5. Just to make sure everything is there, generate any image, then use CLIP interrogator / deepdanbooru in the img2img tab if you need captioning. If you want upscalers, upscale an image in the extra tabs as well. These steps will trigger additional downloads so you need to do them once. I don't remember if you also need to start a dreambooth training, so if you want to be extra sure, create a new model and train it on any image for one epoch.

After these steps, you should have everything you need. You can then either turn off internet access (e.g. using a firewall), or you can also transfer the whole webui directory to another computer that has no internet. For the latter, you will additionally need to copy some contents from the `.cache` folder in your user directory (should be the folders `clip`, `huggingface` and `torch`), and transfer them to the same location on the other computer. Also bear in mind that the webui will need to be at the same path on the other computer, otherwise you will need to repair your `venv` to work at the new path.

If you experience issues with the webui not starting up because it keeps checking for pip updates, you can uncomment that `DREAMBOOTH_SKIP_INSTALL=True` line in the startup bat/sh. But as I said, I think that is not necessary anymore.

Good luck!
